loader/src/LoaderEntry.cpp

```diff
@@ -121,6 +166,109 @@ void mira_escape(struct thread* td, void* uap)
 	printf("[-] mira_escape\n");
 }
 
+#define DIFFERENT_HASH 1
+#define SAME_HASH 0
+
+int MD5_hash_compare(const char *usbfile)
+{
+	 unsigned char c[16];
+	 unsigned char c2[16];
+    int i;
+    FILE *usb = fopen (usbfile, "rb");
+    FILE *hdd = fopen ("/user/MiraLoader.elf", "rb");
+    MD5_CTX mdContext;
+
+    MD5_CTX mdContext2;
+    int bytes2 = 0;
+    unsigned char data2[1024];
```

Comment:
Hash the entire executable, not just the first kb

Reply:
it does hash the whole elf in the while loop 1024 at a time